theatre/core/src/projects/DefaultAssetStorage.ts

```diff
@@ -0,0 +1,172 @@
+import * as idb from 'idb-keyval'
+import type Project from './Project'
+import {val} from '@theatre/dataverse'
+import type {IAssetStorageConfig} from './Project'
+// @ts-ignore
+import blobCompare from 'blob-compare'
+
+function getAllPossibleAssetIDs(project: Project) {
+  const sheets = Object.values(val(project.pointers.historic.sheetsById))
+  const staticValues = sheets
+    .flatMap((sheet) => Object.values(sheet?.staticOverrides.byObject ?? {}))
+    .flatMap((overrides) => Object.values(overrides ?? {}))
+  const keyframeValues = sheets
+    .flatMap((sheet) => Object.values(sheet?.sequence?.tracksByObject ?? {}))
+    .flatMap((tracks) => Object.values(tracks?.trackData ?? {}))
+    .flatMap((track) => track?.keyframes)
+    .map((keyframe) => keyframe?.value)
+
+  const allValues = [...staticValues, ...keyframeValues].filter(
+    // value is string, and is unique
+    (value, index, self) =>
+      typeof value === 'string' && self.indexOf(value) === index,
```

Comment:
how does this filter over `types.string()` values?

Reply:
I found it impossible to filter actual types because these are (afaik) only determined at arbitrary moments during runtime when you call sheet.object(). Filtering for the more general string values is close enough though, and won't break anything.

Reply:
Answered [here](https://github.com/theatre-js/theatre/pull/363#discussion_r1056972282)